tests/components/apcupsd/test_init.py

```diff
@@ -0,0 +1,101 @@
+"""Test init of APCUPSd integration."""
+from collections import OrderedDict
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.apcupsd import DOMAIN, APCUPSdData
+from homeassistant.config_entries import SOURCE_USER, ConfigEntryState
+from homeassistant.const import STATE_UNAVAILABLE
+from homeassistant.core import HomeAssistant
+
+from . import CONF_DATA, MOCK_MINIMAL_STATUS, MOCK_STATUS, init_integration
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.parametrize("status", (MOCK_STATUS, MOCK_MINIMAL_STATUS))
+async def test_async_setup_entry(hass: HomeAssistant, status: OrderedDict) -> None:
+    """Test a successful setup entry."""
+    # Minimal status does not contain "SERIALNO" field, which is used to determine the
+    # unique ID of this integration. But, the integration should work fine without it.
+    await init_integration(hass, status=status)
+
+    # Verify successful setup by querying the status sensor.
+    state = hass.states.get("binary_sensor.ups_online_status")
+    assert state is not None
+    assert state.state != STATE_UNAVAILABLE
+    assert state.state == "on"
+
+
+async def test_multiple_integrations(hass: HomeAssistant) -> None:
+    """Test successful setup for multiple entries."""
+    # Load two integrations from two mock hosts.
+    entries = (
+        await init_integration(hass, host="test1", status=MOCK_STATUS),
+        await init_integration(hass, host="test2", status=MOCK_MINIMAL_STATUS),
+    )
+
+    # Data dict should contain different API objects.
+    assert len(hass.data[DOMAIN]) == len(entries)
+    for entry in entries:
+        assert entry.entry_id in hass.data[DOMAIN]
+        assert isinstance(hass.data[DOMAIN][entry.entry_id], APCUPSdData)
```

Comment:
We should not assert things in hass.data. That's integration details.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Just assert that there are two different config entries setup and that they create different state in the state machine.